### PR TITLE
[PM-33512] feat: Add PremiumStateManager for upgrade banner eligibility

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-33512

## 📔 Objective

Add `PremiumStateManager` that consolidates all premium upgrade banner eligibility logic into a single `StateFlow<Boolean>`. This centralizes the eligibility decision so downstream consumers (e.g., `VaultViewModel`, Settings) observe one signal instead of combining multiple flows themselves.

**Eligibility formula** (all must be true):
- User is not premium
- In-app billing is supported (Play Billing or F-Droid)
- `MobilePremiumUpgrade` feature flag is enabled
- Banner has not been dismissed
- Account is at least 7 days old
- Vault contains at least 5 non-deleted items

**Changes:**
- `PremiumStateManager` interface with `isPremiumUpgradeBannerEligibleFlow` and `dismissPremiumUpgradeBanner()`
- `PremiumStateManagerImpl` combining 5 upstream flows via `combine` with `Clock` injection for account age calculation
- `BillingModule` wiring with `@Provides @Singleton`
- 18 unit tests covering all eligibility conditions, boundary values, `DataState` variants, and reactive updates

### CODE REVIEW

File: app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+/**
+ * Extracts the count of active (non-deleted) vault items from the current [DataState].
+ */
+private fun DataState<VaultData>.activeVaultItemCount(): Int =
+    data
+        ?.decryptCipherListResult
+        ?.successes
+        ?.count { it.deletedDate == null }
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt]: We should eliminate archived items too, right?

File: app/src/test/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImplTest.kt [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+            userId = ACTIVE_USER_ID,
+            isDismissed = true,
+        )
+        val manager = createManager()
+        manager.isPremiumUpgradeBannerEligibleFlow.test {
+            assertFalse(awaitItem())
+        }
+    }
+
+    @Suppress("MaxLineLength")
```
Comment (david-livefront) [inline app/src/test/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImplTest.kt]: We don't need this suppression

File: app/src/test/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+                ),
+            ),
+        )
+        val manager = createManager()
+        manager.isPremiumUpgradeBannerEligibleFlow.test {
+            assertFalse(awaitItem())
+        }
+    }
+
+    @Suppress("MaxLineLength")
```
Comment (david-livefront) [inline app/src/test/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImplTest.kt]: Same here, I'm not gonna leave a comment on all of them though

File: app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    private val unconfinedScope = CoroutineScope(dispatcherManager.unconfined)
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override val isPremiumUpgradeBannerEligibleFlow: StateFlow<Boolean> =
+        combine(
+            authRepository.userStateFlow,
+            billingRepository.isInAppBillingSupportedFlow,
+            featureFlagManager.getFeatureFlagFlow(FlagKey.MobilePremiumUpgrade),
+            authDiskSource.activeUserIdChangesFlow
+                .filterNotNull()
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt]: Will this cause a problem when there is no user since it will never emit?

Comment (david-livefront) [reply app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt]: Could we do this instead?

```kotlin
authDiskSource
    .activeUserIdChangesFlow
    .flatMapLatest { userId ->
        userId
            ?.let { id ->
                settingsDiskSource
                    .getPremiumUpgradeBannerDismissedFlow(id)
                    .map { it ?: false }
            }
            ?: flowOf(false)
    },
```

File: app/src/test/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImplTest.kt (hunk 3 above)
Comment (SaintPatrck) [reply app/src/test/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImplTest.kt]: Gotcha. My bad. Claude hasn't picked up the warning I added about this yet.

File: app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            featureFlagManager.getFeatureFlagFlow(FlagKey.MobilePremiumUpgrade),
+            authDiskSource.activeUserIdChangesFlow
+                .flatMapLatest { userId ->
+                    userId
+                        ?.let { id ->
+                            settingsDiskSource
+                                .getPremiumUpgradeBannerDismissedFlow(id)
+                                .map { it ?: false }
+                        }
+                        ?: flowOf(false)
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt:60]: 👍

File: app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+/**
+ * Extracts the count of active (non-deleted, non-archived) vault items from the
+ * current [DataState].
+ */
+private fun DataState<VaultData>.activeVaultItemCount(): Int =
+    data
+        ?.decryptCipherListResult
+        ?.successes
+        ?.count { it.deletedDate == null && it.archivedDate == null }
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/billing/manager/PremiumStateManagerImpl.kt:119]: 👍
